### PR TITLE
Wrap db usage in try/finally in preprocessing, accept float timestamps

### PR DESCRIPTION
**Pull request by betsybookwyrm**
@SamHames just wanted you to quickly check this isn't going to break anything.

I'm mostly concerned about the int -> float change in pre-processing CSVs. I think a lot of things will generate timestamps with milliseconds?

### CODE REVIEW

File: coordination_network_toolkit/preprocess.py [hunk 1]
```diff
@@ -46,44 +46,46 @@ def preprocess_data(db_path: str, messages: Iterable):
...
+                    db.execute(
+                        "insert or ignore into message_url values(?, ?, ?, ?)",
+                        (message_id, url, timestamp, user_id),
+                    )
 
-    db.execute("commit")
-    db.close()
+        db.execute("commit")
+    finally:
+        db.close()
```
Comment (SamHames) [inline coordination_network_toolkit/preprocess.py:88]: Good catch, I always forget to do this if I don't have a proper context manager for transactions.

File: coordination_network_toolkit/preprocess.py [hunk 2]
```diff
@@ -107,62 +109,64 @@ def preprocess_twitter_json_data(db_path: str, tweets: Iterable[str]):
...
+                tweet["user"]["screen_name"],
+                retweet.get("id_str"),
+                tweet.get("in_reply_to_status_id_str", None),
+                tweet_text,
+                len(tweet_text),
+                zlib.adler32(tweet_text.encode("utf8")),
+                # This will be populated only when similarity calculations are necessary
+                None,
+                # Twitter epoch in seconds
+                int((int(tweet["id"]) >> 22) / 1000),
```
Comment (SamHames) [inline coordination_network_toolkit/preprocess.py:143]: For consistency with the other preprocessing it might make sense to just take the inner float.

**Comment by SamHames** [comment]
I can't think of any way that it would break things. Only issue is if they give a timestamp that is a count of milliseconds instead of a fractional second. Things will still work, it's just that the time_window functionality will be with reference to those units.

**Comment by betsybookwyrm** [comment]
The readme does specify timestamp should be seconds, it just doesn't say it has to be an integer, so I think that's sufficient about the danger of people providing a milliseconds value rather than a seconds value.

File: coordination_network_toolkit/preprocess.py (hunk 1 above)
Comment (betsybookwyrm) [reply coordination_network_toolkit/preprocess.py:88]: I may have had problems parsing my example csv for the library usage example, resulting in db connections left open, and that's how I noticed... 😸

File: coordination_network_toolkit/preprocess.py (hunk 2 above)
Comment (betsybookwyrm) [reply coordination_network_toolkit/preprocess.py:143]: as in just `(int(tweet["id"]) >> 22) / 1000` without the outer int()?

Comment (betsybookwyrm) [reply coordination_network_toolkit/preprocess.py]: Made the change.

Comment (SamHames) [reply coordination_network_toolkit/preprocess.py]: Yep that's what I meant :)

**Pull request merged**